plugin/caffe/caffe_loss-inl.h

```diff
@@ -141,7 +141,7 @@ class CaffeLoss : public Operator {
     top_[0]->set_cpu_diff(&grad_scale_);
 
     // Set BP flag
-    for (index_t i = 0; i < param_.num_data; ++i)
+    for (int i = 0; i < param_.num_data; ++i)
       flags_[i] = req[i] != kNullOp;
 
     caffeOp_->Backward(top_, flags_, bot_);
```

Comment:
@piiswrong I checked all layers with any cpu operation in gpu code and indeed found one case, i.e. forward_gpu() of ReductionLayer, that needs to call gpu_data/diff to sync.
I'll add cpu->gpu data/diff sync at the end of both forward and backward.